src/shared/dissect-image.c

```diff
@@ -1661,6 +1701,40 @@ int dissect_image_and_warn(
         }
 }
 
+bool dissected_image_can_do_verity(const DissectedImage *image, unsigned partition_number) {
+        int k;
+
+        k = PARTITION_VERITY_OF(partition_number);
+        if (k >= 0)
+                return true;
+        if (image->can_verity) {
+                /* image->can_verity is meaningful on a per-partition check only with a single file system */
+                for (unsigned i = PARTITION_ROOT + 1; i < _PARTITION_DESIGNATOR_MAX; i++)
+                        if (image->partitions[i].found)
+                                return false;
```

Comment:
Hmm, I think it would be prettier to just store a boolean flag `naked_file_system` or so in the `DissectImage` structure that indicates whether we have a partition table or not, instead of checking this after the fact. After all this code could not distuingish the case "gpt with only one root partition" from "no partition table". This check then becomes a lot easier:

```
if (image->naked_file_system)
        return designator == PARTITION_ROOT && image->can_verity;

return PARTITION_VERITY_OF(designator) >= 0;
```

Reply:
Ok, no strong preference - done